### PR TITLE
Add deleteStream function to RTMPStream

### PR DESCRIPTION
**Pull request by talhatsk**
## Description & motivation

This PR adds a `deleteStream` function to the `RTMPStream` class, similar to the `closeStream` function. It ensures the stream is properly deleted when stopping.

### Changes Made
- Added `deleteStream` method to `RTMPStream.swift`.
- Updated the `close` method to call `deleteStream` after closing the stream.

## Type of change
Please delete options that are not relevant.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: Sources/RTMP/RTMPStream.swift [hunk 1]
```diff
@@ -489,31 +493,56 @@ open class RTMPStream: IOStream {
...
+                                    streamId: RTMPChunk.StreamID.command.rawValue,
+                                    message: RTMPCommandMessage(
+                                        streamId: 0,
+                                        transactionId: 0,
+                                        objectEncoding: self.objectEncoding,
+                                        commandName: "closeStream",
+                                        commandObject: nil,
+                                        arguments: [self.id]
+                                    )))
+        deleteStream()
```
Comment (shogo4405) [inline Sources/RTMP/RTMPStream.swift:520]: calling deleteStream after calling closeStream. I am unsure about the validity of this and have concerns about potential side effects. Could you tell me what tests you conducted

Comment (xoraingroup) [reply Sources/RTMP/RTMPStream.swift:520]: We should separate close from deleteStream completely. I agree with @shogo4405 , there may be side effects of this.

Comment (talhatsk) [reply Sources/RTMP/RTMPStream.swift]: removed it deleteStream() from close function and add a separate function for it

File: Sources/RTMP/RTMPStream.swift [hunk 2]
```diff
@@ -489,31 +493,58 @@ open class RTMPStream: IOStream {
     }
 
     func close(withLockQueue: Bool) {
-        if withLockQueue {
-            lockQueue.sync {
-                self.close(withLockQueue: false)
+            if withLockQueue {
```
Comment (levs42) [inline Sources/RTMP/RTMPStream.swift:496]: Something happened to the formatting

**Comment by shogo4405** [comment]
Since it seemed like the source code review would take some time, I went ahead and did it myself. https://github.com/shogo4405/HaishinKit.swift/pull/1517

**Pull request closed without merging**